spec/credit_card_validations_spec.rb

```diff
@@ -92,66 +91,74 @@
   end
 
-  it "should  be valid? if right brand" do
+  it 'should  be valid? if right brand' do
     detector(VALID_NUMBERS[:visa].first).valid?(:mastercard, :visa).must_equal true
     detector(VALID_NUMBERS[:visa].first).valid?(:mastercard, :amex).must_equal false
   end
 
 
-  describe "adding/removing brand" do
-
-
-    let(:voyager_number) {
-      '869926275400212'
-    }
-
-    it "should validate number as voyager" do
-      CreditCardValidations::Detector.add_brand(:voyager, {length: 15, prefixes: '86'})
-      detector(voyager_number).valid?(:voyager).must_equal true
-      detector(voyager_number).voyager?.must_equal true
-      detector(voyager_number).brand.must_equal :voyager
-    end
-
+  describe 'adding/removing brand' do
 
-    describe "Add voyager rule" do
-      before do
-        CreditCardValidations::Detector.add_brand(:voyager, {length: 15, prefixes: '86'})
-      end
+    describe 'adding rules' do
 
-      it "should validate number as voyager" do
+      let(:voyager_number) { '869926275400212' }
 
+      it 'should validate number as voyager' do
+        CreditCardValidations::Detector.add_brand(:voyager, length: 15, prefixes: '86')
         detector(voyager_number).valid?(:voyager).must_equal true
         detector(voyager_number).voyager?.must_equal true
         detector(voyager_number).brand.must_equal :voyager
-
       end
 
-      describe "Remove voyager rule" do
+      describe 'Add voyager rule' do
         before do
-          CreditCardValidations::Detector.delete_brand(:voyager)
+          CreditCardValidations::Detector.add_brand(:voyager, length: 15, prefixes: '86')
         end
 
-        it "should not validate number as voyager" do
-          detector(voyager_number).respond_to?(:voyager?).must_equal false
-          detector(voyager_number).brand.must_be_nil
+        it 'should validate number as voyager' do
+          detector(voyager_number).valid?(:voyager).must_equal true
+          detector(voyager_number).voyager?.must_equal true
+          detector(voyager_number).brand.must_equal :voyager
         end
-      end
 
-    end
+        describe 'Remove voyager rule' do
+          before do
+            CreditCardValidations::Detector.delete_brand(:voyager)
+          end
 
+          it 'should not validate number as voyager' do
+            detector(voyager_number).respond_to?(:voyager?).must_equal false
+            detector(voyager_number).brand.must_be_nil
+          end
+        end
+      end
+    end
 
-    it "should raise RuntimeError" do
-      proc { CreditCardValidations::Detector::add_rule(:undefined_brand, 20, [20]) }.must_raise RuntimeError
+    describe 'plugins' do
+      [:diners_us, :en_route, :laser].each do |brand|
+        it "should support #{brand}" do
+          -> { CreditCardValidations::Factory.random(brand) }.
+            must_raise(CreditCardValidations::Error)
+          detector("somenumber").respond_to?("#{brand}?").must_equal false
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.